Library/Homebrew/test/rubocops/cask/stanza_order_spec.rb

```diff
@@ -459,39 +459,22 @@
     include_examples "autocorrects source"
   end
 
-  # TODO: detect out-of-order stanzas in nested expressions
-  context "when stanzas are nested in a conditional expression" do
-    let(:source) do
-      <<~CASK
-        cask 'foo' do
-          if true
-            sha256 :no_check
-            version :latest
-          end
-        end
-      CASK
-    end
-
-    include_examples "does not report any offenses"
-  end
-
   context "when `on_arch` blocks are out of order" do
     let(:source) do
       <<~CASK
         cask 'foo' do
           on_intel do
-            url "https://foo.brew.sh/foo-intel.zip"
-            sha256 :no_check
             version :latest
-          end
+            sha256 :no_check
 
+            url "https://foo.brew.sh/foo-intel.zip"
+          end
           on_arm do
-            url "https://foo.brew.sh/foo-arm.zip"
-            sha256 :no_check
             version :latest
-          end
+            sha256 :no_check
 
-          name "Foo"
+            url "https://foo.brew.sh/foo-arm.zip"
+          end
```

Comment:
So this is only testing the order of `on_arm` and `on_intel`, and the one below tests only the order of inner stanzas. Should we have another test which tests both, as this did before, or merge these into one that tests both?

Reply:
Ahhh, yes, this new test is important!

```
  1) RuboCop::Cop::Cask::StanzaOrder when the `on_os` stanzas and their contents are out of order autocorrects source
     Failure/Error: raise Parser::ClobberingError, "Parser::Source::TreeRewriter detected clobbering" if action == :raise
     
     Parser::ClobberingError:
       Parser::Source::TreeRewriter detected clobbering
     Shared Example Group: "autocorrects source" called from ./test/rubocops/cask/stanza_order_spec.rb:561
```

Reply:
Hmmm, I see, it detects the toplevel stanzas (`on_mojave`) first, adds those offenses, then descends into the `on_mojave` block and detects those, then tries to correct them all but the ordering means that they all move about hence the "clobbering" from what it originally detected? Those are probably some bad words for explaining what's happening, but that's all I can come up with right now. No idea how to fix it, either.

I tried doing the _inner_ stanzas first but that doesn't help, but maybe I'm doing it wrong. 🤔